integrations/cdi/datasource-ucp/src/main/java/io/helidon/integrations/datasource/ucp/cdi/UCPBackedDataSourceExtension.java

```diff
@@ -223,6 +223,10 @@ private static PoolDataSourceImpl createDataSource(final Instance<Object> instan
             }
             final Object serviceName = connectionFactoryProperties.remove("serviceName");
             final Object pdbRoles = connectionFactoryProperties.remove("pdbRoles");
+            // Used for OCI ATP Integration
+            // Removing this so that it is not set on connectionFactoryProperties,
+            // Else we get exception with getConnection using this DS, if its set.
+            connectionFactoryProperties.remove("tnsNetServiceName");
```

Comment:
Let me think about a better way to do this so there is a minimum of "magic" properties in the generic extension.

Reply:
Looks like there's not given that `tnsNetServiceName` is not a Java Bean property recognized by the underlying data source and that it (I guess) throws an `IllegalArgumentException` in the presence of bad properties here, which normally would be a feature, not a bug, but if we were previously relying on this…anyway, carry on.

Reply:
Thanks @ljnelson for the review. Will look to see if adding observer for PoolDataSource in ATP integration can remove this special handling in extension.